File: 0704-binary-search/0704-binary-search.py
class Solution:
    def search(self, nums, target):
        left = 0
        right = len(nums)-1
        while left <= right:
            mid = (left + right) // 2
            if nums[mid] == target:
                return mid
            if nums[mid] < target:
                left = mid+1
            if nums[mid] > target:
                right = mid-1
        return -1

File: 0704-binary-search/0704-binary-search.cpp
class Solution
{
public:
    int search(vector<int> &nums, int target)
    {
        int left = 0,mid;
        int right = nums.size() - 1;
        while (left <= right)
        {
            mid = left + (right - left) / 2;
            if (nums[mid] == target)
                return mid;
            if (nums[mid] < target)
                left = mid + 1;
            else
                right = mid - 1;
        }
        return -1;
    }
};